scripts/process_uploads.py

```diff
@@ -3,95 +3,151 @@
+                        sample_id_internal=internal_sample_id,
+                        data_file=FileGroup(
+                            path=data_file_path, basename=data_file_basename
+                        ),
+                        index_file=FileGroup(
+                            path=index_file_path, basename=index_file_basename
+                        ),
+                        md5=FileGroup(
+                            path=data_file_path + '.md5',
+                            basename=data_file_basename + '.md5',
+                        ),
+                        batch_number=batch_number,
+                    )
+
+                    archive_files.append(sample_group_archive)
 
-    all_samples, csv_path_upload = samples_from_csv(upload_bucket, upload_prefix)
-    previous_samples, _ = samples_from_csv(metadata_bucket, metadata_prefix)
+    return main_files, archive_files
 
-    samples = set(all_samples) - set(previous_samples)
 
-    curr_csv_file_path = join(upload_bucket, csv_path_upload)
+def setup_job(
+    batch: hb.batch, name: str, docker_image: Optional[str], python_job: bool = False
+) -> hb.batch.job:
+    """ Returns a new Hail Batch job that activates the Google service account. """
 
-    return samples, curr_csv_file_path
+    job = batch.new_python_job(name=name) if python_job else batch.new_job(name=name)
 
+    if docker_image is not None:
+        job.image(docker_image)
 
-def generate_file_list(
-    samples: Set[str],
-) -> Tuple[List[SampleGroup], List[SampleGroup]]:
-    """ Generate list of expected files, given a list of samples """
-    main_files = []
-    archive_files = []
-    for s in samples:
-        sample_group_main = SampleGroup(
-            f'{s}.g.vcf.gz', f'{s}.g.vcf.gz.tbi', f'{s}.g.vcf.gz.md5'
-        )
-        main_files.append(sample_group_main)
-        sample_group_archive = SampleGroup(
-            f'{s}.cram', f'{s}.cram.crai', f'{s}.cram.md5'
+    if not python_job:
+        job.command(
+            'gcloud -q auth activate-service-account --key-file=/gsa-key/key.json'
         )
-        archive_files.append(sample_group_archive)
 
-    return main_files, archive_files
+    return job
 
 
-def setup_job(batch: hb.batch, name: str, docker_image: Optional[str]) -> hb.batch.job:
-    """ Returns a new Hail Batch job that activates the Google service account. """
+def create_analysis_in_sm_db(sample_group: SampleGroup, proj, path, analysis_type):
+    """ Creates a new analysis object"""
+    aapi = AnalysisApi()
 
-    job = batch.new_job(name=name)
-    if docker_image is not None:
-        job.image(docker_image)
+    internal_id = sample_group.sample_id_internal
 
-    job.command('gcloud -q auth activate-service-account --key-file=/gsa-key/key.json')
+    if analysis_type == 'gvcf':
+        file_extension = '.g.vcf.gz'
+    else:
+        file_extension = '.cram'
 
-    return job
+    filepath = os.path.join('gs://', path, internal_id + file_extension)
```

Comment:
Can you pass the fully formed gs path to the file into this method, as "output_path" instead of constructing it inside, and outside. If you change the format in one place, it would be difficult to remember to change the other.